src/api/app/controllers/staging_project_controller.rb

```diff
@@ -0,0 +1,10 @@
+class StagingProjectController < ApplicationController
```

Comment:
Why do you need the inheritance? Just put everything into StagedRequestsController?

Reply:
> I would prefer like that since we are going to need it to different sub controllers too.

If we need it in the future we can introduce it in the **future**, no? Program for today, not the future :smile: 

![selection_002](https://user-images.githubusercontent.com/3799140/47660229-e5b19d00-db96-11e8-88ed-b59e56a26ee4.png)

Reply:
oh well, can we please concentrate in what is important to delivery it? Thank you.

Reply:
> can we please concentrate in what is important to delivery it?

I think I do :thinking: You asked for my review and here it is, if you don't like it... :smile: Go ahead and merge if you think it is not important ...

Reply:
![screenshot_2018-10-29_16-52-39](https://user-images.githubusercontent.com/37418/47662398-1e537580-db9b-11e8-9aa7-003882fd7b7b.png)

waiting to go back to test unit because dhh said it

Reply:
I'd appreciate if you don't take reviews personal :thinking: You asked for my review, this is my opinion which I showed with an example from dhh. The purpose of a review is not to confirm you in your decisions but also point out / discuss other opinions.

I approved the PR, just go ahead and merge if you think it is not a valid point.

Reply:
And can we maybe not delete comments :wink:

Reply:
actually i deleted it as accident.

Reply:
> I'd appreciate if you don't take reviews personal  You asked for my review, this is my opinion which I showed with an example from dhh.

Personal would be if I think that's against me. I don't think so. My first push was everything in one controller, and you asked to change it to structure the code like it is now, to be better adaptable to the future. So I did it, and code that I know that's shared, I moved to a base controller. That's OO principle applied to controllers, not trying to predict the future. 

What I do have problem with is with distracting micromanagement in PR and with distracting discussions full of passive aggressiveness

Reply:
> You asked for my review, this is my opinion which I showed with an example from dhh

if dhh quotes were critical, we wouldnt be using rspec or haml. I didn't ask your review, you were already the reviewer.

Reply:
> What I do have problem with is with distracting micromanagement in PR and with distracting discussions full of passive aggressiveness.

If any of my comments were passive aggressive I'm sorry, this was not my intention. My intention is to improve the code and I still don't understand how this is micromanaging.

> The first push was everything in one controller, and you asked to change it to structure the code like it is now, to be better adaptable to the future. So I did it, and code that I know that's shared, I moved to a base controller. That's OO principle applied to controllers, not trying to predict the future.

Actually this is exactly predicting the future because we just don't know what code will be shared in the future.

> if dhh quotes were critical, we wouldnt be using rspec or haml. 
Not everything is black and white :smile: Some persons have valid opinions which does not mean all opinions of this person is valid :smile: As said, I just used the tweet as an example.